blocklib/digitizers/lib/app_buffer.h

```diff
@@ -170,68 +175,27 @@ class app_buffer_t
         d_data_rdy_cv.notify_one();
     }
 
-    /*!
-     * \brief This method follows the GR's work method signature/approach, that is the
-     * pointers to the GR output buffers should be passed in.
-     *
-     * NOTE, clients MUST call wait_data_ready before attempting to invoke this method.
-     *
-     * Returns number of data chunks lost from the last call.
-     */
-    int get_data_chunk(std::vector<float*>& ai_buffers,
-                       std::vector<float*>& ai_error_buffers,
-                       std::vector<uint8_t*>& port_buffers,
-                       std::vector<uint32_t>& status,
-                       int64_t& local_timestamp)
+    data_chunk_ptr get_data_chunk()
```

Comment:
"I suppose it's the callers' duty to call the release method after usage." data_chunk_ptr is a unique_ptr doing the release on destruction (not to be confused with the private data_chunk_sptr...), but yes, I should document that. (goes for all your similar comments about documentation as well :) )

Reply:
Thank you, much appreciated !

Reply:
About the mismatch between timing messages and triggers (because triggers are missed): According to Ralph, this shouldn't be an issue in the new implementation, so I'm closing this as resolved.

Reply:
Not clear to me how you want to guarantee that / I would not place any bet on it … though, well, I am not the one to pay the bills, so let's move on.